### PR TITLE
fix(setup): Zope root cookie auth and login form

### PR DESCRIPTION
**Pull request by rpatterson**
The GenericSetup "various" import step that originally installs PAS into a Plone portal
also migrates the Zope root `/acl_users`.  Probably an accident over time, but it
results in a cookie auth plugin that doesn't work outside of the Plone portal:

    2021-12-27 11:12:02,243 ERROR   [Zope.SiteErrorLog:22][waitress-0] ComponentLookupError: http://localhost:49080/api/acl_users/credentials_cookie_auth/login
    Traceback (innermost last):
      Module ZPublisher.WSGIPublisher, line 162, in transaction_pubevents
      Module ZPublisher.WSGIPublisher, line 372, in publish_module
      Module ZPublisher.WSGIPublisher, line 266, in publish
      Module ZPublisher.mapply, line 85, in mapply
      Module ZPublisher.WSGIPublisher, line 63, in call_object
      Module Products.PlonePAS.plugins.cookie_handler, line 106, in login
      Module Products.PluggableAuthService.PluggableAuthService, line 1153, in updateCredentials
      Module Products.PlonePAS.plugins.cookie_handler, line 74, in updateCredentials
      Module zope.component._api, line 165, in getUtility
    zope.interface.interfaces.ComponentLookupError: (<InterfaceClass plone.registry.interfaces.IRegistry>, '')

This import step also removes the `login_form` template which breaks the challenge
response.

Add an interface check to decide whether to install Plone's `ExtendedCookieAuthHelper`
or PAS's vanilla `CookieAuthHelper`.

### CODE REVIEW

**Comment by rpatterson** [comment]
@jenkins-plone-org please run jobs

**Comment by rpatterson** [comment]
Everything is green, @jensens.  LMK if there's anything I can do to move this forward.

File: src/Products/PlonePAS/setuphandlers.py
Comment (jensens) [inline src/Products/PlonePAS/setuphandlers.py:221]: f-strings preferred in Python 3 land. Anyway, minor one.
```suggestion
        logger.debug(
            f"Added {cookie_meta_type}: "
            f"{'/'.join(uf.credentials_cookie_auth.getPhysicalPath()}"
        )
```

**Review by jensens — APPROVED** [review]
Overall LGTM. Suggested minor change would be nice to get rid of old format strings.

File: src/Products/PlonePAS/setuphandlers.py
Comment (rpatterson) [reply src/Products/PlonePAS/setuphandlers.py:221]: I've still been using `%` string formatting for logging calls, @jensens, because the arguments passed into the `logger.{level}(msg, args)` are actually recorded in the `args` attribute on the `LogRecord` which may be used by `Handler`s, `Filter`s, etc..  Very minor issue, but it also postpones interpolation until after decisions have been made about whether to emit a given record.  IOW, under normal operation the interpolation for a `DEBUG` level record such as this case will never even happen if we use `%` string formatting but would always happen if we use `f"..."` strings.

So I prefer keeping it as is.  If you still disagree and this will block merging, feel free to unresolve and LMK and I'll make and push the change.

**Comment by rpatterson** [comment]
LMK if I should merge this myself or if there's otherwise anything I should do to get this merged, @jensens.

**Comment by Rudd-O** [comment]
Wow, this is astonishingly good work.  Not only is the code clearer, it must have taken a helluva lot of work to understand the old one (in particular with the amount of context needed from back in the Zope 2 days) and get it working again.  Thanks.

File: src/Products/PlonePAS/setuphandlers.py
Comment (rpatterson) [reply src/Products/PlonePAS/setuphandlers.py:221]: Just to capture some other context here in case anyone else runs across this, I recently ran across [the `Use %s formatting in logging functions` warning from PyLint](https://pycodequ.al/docs/pylint-messages/w1203-logging-fstring-interpolation.html) which both demonstrates agreement from at least some part of the Python code quality community and points out another reason not to use f-strings for logging calls, namely lazy evaluation.  IOW, the work of interpolating strings isn't done *unless* the given logging record will actually be logged.  For example, one can feel free to do a bunch of rich interpolation in `DEBUG` records without fear that they'll incur a performance penalty under normal operation.

**Pull request merged**
